### ISSUE TITLE
How do people get the modern, fixed version? And what is the long term outlook for the module?

### ISSUE THREAD

**Issue by fake-name** [VeNoMouS/cloudscraper#241]
From various other issue reports:

 - https://github.com/VeNoMouS/cloudscraper/issues/238
 - https://github.com/VeNoMouS/cloudscraper/issues/236
 - https://github.com/VeNoMouS/cloudscraper/issues/209

It seems that this project has gone private/paid. 

That's understandable, through frustrating for people (like myself) who have previously donated and written code to integrate cloudscraper. 

Given this, it seem to me that the fact that the version here doesn't work should be made a lot more apparent. Ideally, it seems like it should probably be unpublished from PyPi since the versions there aren't functional, and the homepage readme should feature this front and foremost. 

Additionally, for people willing to buy it, how the hell do you do so? There's random mentions of discord, but I don't use and am uninterested in trying to navigate obscure obnoxious forums and I particularly dislike discord in particular (if your web forum requires me to enable client-side JS, it can go fuck itself). 

So basically:

 - What's the path to actually purchasing?
 - What's the price?
 - Do previous donations apply?
 - What is the long term plan? 
 - How do paid modules even work?
 - Is there a private PyPi-like facility for managing installation? 

TL;DR fuck Cloudflare for ruining the internet. There should be no context where I need to execute JS to issue a HTTP GET.

**Comment by timb699** [comment]
> the homepage readme should feature this front and foremost
I agree with you here.

> What's the path to actually purchasing?
You're apparently supposed to message VeNoMouS on Discord and he sends you prices. I've sent him a message and am awaiting a reply

> What's the price?
Someone told me that these are the prices:
![image](https://user-images.githubusercontent.com/59662201/84637449-9c3dcf00-aeed-11ea-90e7-20a1b4808876.png)

I don't know the answers to the rest of your questions unfortunately, and am waiting to find out too.

**Comment by bjgood** [comment]
Is that pricing in euros per month?

**Comment by timb699** [comment]
By the looks of it yes

**Comment by bjgood** [comment]
Thanks for the reply, timb699.
Seems terribly pricey for a single user. Looks like cfscrape support is dead for an average user with these prices.
Later...

**Comment by 102flex** [comment]
> Thanks for the reply, timb699.
> Seems terribly pricey for a single user. Looks like cfscrape support is dead for an average user with these prices.
> Later...

15 € per month are too expensive? There are certainly several days of work involved and you also need a lot of previous knowledge. + Running costs for the owner. 

I can't relate that.

**Comment by bjgood** [comment]
@102flex.
To each his own, I guess. I may change my mind later on, but as of right now, it's way more than I'm willing to pay for just the results of a few scrapers.
Later...

**Comment by cryptocake** [comment]
What are you waffling? You also get free groceries? Cab driver gets you home without a charge too yea? This man is working his ass off and you bitch about 15 euro's?
![image](https://user-images.githubusercontent.com/15063245/84709246-2758ac80-af62-11ea-85c0-dfba2e256f7e.png)

**Comment by bjgood** [comment]
@cryptocake.
I know Venomous works hard on this project, and I appreciate the effort he puts into it. But, I think the price is too much. I have the right to express my opinion on this subject, you know. And, I'm not going the fuck anywhere (especially after that moronic png you posted). I suggest therapy for your idiot impulses. Good luck with that.
Later...

**Comment by VeNoMouS** [comment]
So while you appreciate all the work and hours I put into the project with constant patching and on going investigation each time cloudflare change their challenges,  and/or helping users when they cant work out a problem...

All those man hours should be free and on my back?? what a world you must live in... glad you don't pay for rent / power / utilities / tv subscriptions.... i hope you dont take home a pay check and give your employer all that work for free..

**Comment by bjgood** [comment]
Yes, I suppose you do look like when you're mad. :). But, with all the thousands of individuals in the world who need cfscrape functioning properly times almost 17 USD per individual per month...that's a bit more than rent, power, tv subs, etc.. And, I didn't say you should do it for free. Please keep your rant in perspective, if you don't mind. Just charge something a little bit more inline with the level of demand.
I have the greatest respect for your abilities, Venomous, and all that you have done in the past. I'm just expressing my opinion.
Sorry if it pissed you off...which I obviously have.
This is all I have to say on the subject.
Later...

**Comment by timb699** [comment]
@bjgood In my opinion it isn't expensive especially considering the fact that it requires daily maintenance, in fact it seems pretty cheap if you think of the work he does. Obviously you have your own opinion and if you feel that it's too pricey, then just don't get it.

@VeNoMouS Sorry to tag you, but when you find time, can you please drop me the purchase link to cloudscraper on Discord or reply to me or whatever. Thanks for providing us cloudscraper!

**Comment by ctrl-i** [comment]
@bjgood couldn't agree more with you.  The abuse you are getting for giving your opinion is not warranted as you are allowed an opinion.  If someone replied to me like they did to you I wouldn't do business with them at any price.

**Comment by cryptocake** [comment]
Why do you even want to give an "opinion" here in a github repository? This isn't social media, nor a democracy. Either you contact @VeNoMouS to purchase, or you _simply leave_ and go build your own.

**Comment by ctrl-i** [comment]
I built my own, took a couple of days.  Certainly wasn't number of subscribers x €17 a month worth of effort.  Maybe you are going about it the wrong way.

Why are you even wanting to run a business from a FREE github repository?!  Are you too cheap to pay for hosting?!

**Comment by lilyoungbug** [comment]
If Venom would have made the “new cloudscraper” public cloudflare would have patched instantly. Venom is a great dev + he’s spending several hours on this repository. and i think that 15€+ is more than fair. You are paying for his support and his constant updates. If you are complaining about 15€ then you are living in another world. I think that if you work on something (nothing that big for example) you cannot assume to not getting paid. Cheers, feel free to express your opinion but do it somewhere else.

**Comment by VeNoMouS** [comment]
![giphy](https://user-images.githubusercontent.com/597545/84759524-f4beba80-b01a-11ea-9d57-9dba6599e4a8.gif)

**Comment by timb699** [comment]
@VeNoMouS Is there a link to purchase or do we just wait for a Discord reply/server invite? Sorry but this purchase procedure is confusing me a bit. Thanks again

**Comment by VeNoMouS** [comment]
@timb699 I havent put up an order page yet, purely to get ppl on to iron out bugs, most ppl who have been msging me on discord ive been giving free trials to

**Comment by timb699** [comment]
@VeNoMouS oh alright. So I just wait for a Discord reply. Thank you

**Comment by samuel-obiedzinski** [comment]
@VeNoMouS

Please correct me if I am wrong - does this means that you send JS to your server where you solve the task and resend it back to client?
So, the JS Challenge is not calculated on client device? Am I right?

Does this mean that you are using regular web browser which is running on your server to calculate js challenge?

**Comment by VeNoMouS** [comment]
Yes, I solve all the challenges on my end... its a SaaS , all requests are made by you the client, i dont make a single call to cloudflare

```
Does this mean that you are using regular web browser which is running on your server to calculate js challenge?
```

LOL NO, god no

**Comment by ddm4313** [comment]
> Yes, I suppose you do look like that when you're mad. :). But, with all the thousands of individuals in the world who need cfscrape functioning properly times almost 17 USD per individual per month...that's a bit more than rent, power, tv subs, etc.. And, I didn't say you should do it for free. Please keep your rant in perspective, if you don't mind. Just charge something a little bit more inline with the level of demand.
> I have the greatest respect for your abilities, Venomous, and all that you have done in the past. I'm just expressing my opinion.
> Sorry if it pissed you off...which I obviously have.
> This is all I have to say on the subject.
> Later...
How is 17 usd more than rent, power, tv subs? It's really nothing..

**Comment by ddm4313** [comment]
Bad service, denies access even though I wanted to pay, dead project

**Comment by timb699** [comment]
@ddm4313 denies access to what?

**Comment by VeNoMouS** [comment]
He got denied into the trial group so he had a hissy

**Comment by VeNoMouS** [comment]
@timb699 what is your discord username? i dont see a msg from you..

**Comment by timb699** [comment]
@VeNoMouS it's timb699#7058

**Comment by KyranRana** [comment]
@VeNoMouS Good job man. Managed to solve v2?

**Comment by samuel-obiedzinski** [comment]
@VeNoMouS

Thank you for the answer. What about captcha challenge? Do your service bypass it too or maybe captcha challenge must be handled by other service like 2captcha?

**Comment by vrikshasana** [comment]
Doesn't work for me. ,
Found analysis of the new challenge here
https://www.linkedin.com/pulse/cf-ddos-challenge-behavior-asaf-even-tzur/?trackingId=PRL5lZUCTFC49Y4MRSodKQ%3D%3D

**Comment by VeNoMouS** [comment]
@vrikshasana uhh..... thats a really badly written write up on the v1 challenge, which doesnt take into account half variances that are in v1.

**Comment by Seraphppy** [comment]
You guys needs to chill, if you dont like the price move on. Im on the trail and its working v well!

**Comment by UrbiJr** [comment]
Seems like a point of view from an insider is due here..
I'm part of the trial group of the new V2 solver, which venom obviously had to **limit** by creating subscriptions, because he's got cf devs constantly monitoring him..

Anyways, I must say my experience with the new SaaS solver is just flawless so far. Venom is doing a great job behind the scenes. bugs are getting reported and updates are released for both win & unix.
You won't have to change your code, you're only required to add an additional parameter when creating your scraper. 
Can't say anything bad about it, nor the licensing, for which I was only required to include a license.lic file inside my project folder, and install pyarmor (this will be explained better once you get your subscriptions, but it really takes 2 mins)

That being said, please gentlemen _stop_ wasting venom's **precious** time when there's already all you need to know...

**Comment by fake-name** [comment]
Considering my usage was entirely for personal hobby projects, 15 euros a month is pretty steep.

Also, as someone who actively dislikes discord, hasn't used it, and can potentially offer actual debugging support, what would be the path to potentially purchasing?

As an aside, I feel a "not for commercial use" licence variant would be nice, but I also think enough of the people who need to get through cloudflare (particularly people who are doing it for money-making reasons) are probably skeezy enough that it'd not be reasonable ti expect people to not lie when purchasing.

**Comment by samuel-obiedzinski** [comment]
@fake-name

> Edit: I just realized, depending on how things work, you could potentially audit the solved URLs that are CF wrapped.

Yep, this is potential security issue. For example when login site is protected by Cloudflare and login and password are send using GET method then these data goes in plain form to the @VeNoMouS ...

**Comment by VeNoMouS** [comment]
The only data that goes to me is the challenge data, but ... guess what ... simply don’t use my product, problem solved

**Comment by VeNoMouS** [comment]
Lets make life easier for everyone, any further whinging and whining about stuff I have asked to be discussed on discord is just gonna end up getting you banned from my github, i am tired of the drama...

**Comment by samuel-obiedzinski** [comment]
@VeNoMouS

To solve challenge you need to have full request url. Because some task in CF challenge checks it.

**Comment by evanspy1** [comment]
Just an FYI, you have officially been "noticed"...by the CEO of CF.  Have fun with that.

https://twitter.com/eastdakota/status/1273277839102656515

**Comment by jgrahamc** [comment]
🍿

**Comment by 2020iscancelled** [comment]
Now that we've got your attention Mr. Prince:
Thank you Cloudflare for your dedication to ruining the internet.

**Comment by evanspy1** [comment]
Yes thank you very much ^^^

**Comment by Goldenpistol** [comment]
@VeNoMouS how can I get your discord id?

**Comment by VeNoMouS** [comment]
Dear Cloudflare,

`B tusbohf Hbnf. Uif pomz xjoojoh npwf jt opu up qmbz.` 

`Xjuipvu vt, zpv xpvmeo'u fyjtu, uif ofyu npwf jt zpvst.`

**Comment by VeNoMouS** [comment]
Dear Cloudflare,

```
B tusbohf Hbnf. Uif pomz xjoojoh npwf jt opu up qmbz.
```

```
Xjuipvu vt, zpv xpvmeo'u fyjtu, uif ofyu npwf jt zpvst.

CUX, Hsfbu npwjf... Tibmm xf qmbz b hbnf?
```

**Comment by rubber777** [comment]
@VeNoMouS Hi. Please check your discord friend requests / messages. Would need to buy the module urgently until tomorrow if possible..

**Comment by jgrahamc** [comment]
![IMG_5267](https://user-images.githubusercontent.com/53644/85679245-242b9200-b6c1-11ea-9b74-37f89c9b0bd3.jpg)

**Comment by zedd3v** [comment]
WOW! Caesar cipher. I never expected Cloudflare to be run by anonymous.
At least my bypass still works https://youtu.be/dQw4w9WgXcQ

**Comment by ngosang** [comment]
> frankly if you don't like even the cheapest option i'm requesting, you are more than welcome to make your own solver and give it out to everyone and do free on-going support.

Free 4 all (it's experimental, any help is welcome) => https://github.com/ngosang/FlareSolverr
Related info => https://github.com/Jackett/Jackett/issues/9029

**Comment by zedd3v** [comment]
Too slow and not reliable

**Comment by VeNoMouS** [comment]
just like the owner of this project...

![image](https://user-images.githubusercontent.com/597545/85804974-2a903b80-b79f-11ea-951c-4d75594fb5c2.png)

**Comment by forty875** [comment]
@VeNoMouS
added you on discord please check... My username is forty2877 .

**Comment by fake-name** [comment]
Why was this closed? There's still no information in the readme about any of the questions I asked.

**Comment by VeNoMouS** [comment]
@jgrahamc we just wanted to thank you and the rest of your team for the batman easter egg, put a smile on our faces :)

**Comment by zedd3v** [comment]
@jgrahamc I see your "techleads" fell for my rickroll URL.
You also got me ):

**Comment by VeNoMouS** [comment]
@jgrahamc 
![image](https://user-images.githubusercontent.com/597545/88011048-72ba3800-cb6a-11ea-8a1d-777d7844540b.png)

**Comment by resynth1943** [comment]
> Just an FYI, you have officially been "noticed"...by the CEO of CF. Have fun with that.
> 
> https://twitter.com/eastdakota/status/1273277839102656515

Wait, what the fuck?

I didn't realise that was the CEO, Jesus. That's unprofessional.

**Comment by VeNoMouS** [comment]
It's a business, it looks good for him making us look bad... take it on the chin and not to the heart, move on :)

**Comment by resynth1943** [comment]
> It's a business, it looks good for him

I wouldn't say it looks good at all. At best, it's unprofessional to the extreme.
